### DISCUSSION TITLE
Questions on Predictions and Training on Sleap

### DISCUSSION THREAD

**Discussion by rhansen123** [talmolab/sleap#1564] [Help!]
Hello! So I am new to using SLEAP and have been training a model in order to analyze some data my lab will be collecting in another week or so. I first put instances on 1,000 random frames (total) from 55 videos. I then trained and fixed 250 predictions and would train again, because fixing 1,000 predictions seemed excessive. However, I noticed that old predictions seemed to stay in addition to the new predictions with each trained model. It seems like a waste of time to fix/label 1,000 frames and then train five or more times. I am curious for advice on how to go about this?

**Comment by roomrys** [comment]
Hi @rhansen123,

Sorry, I am a bit confused - I think there might be a jargon mix up here, but I'll try to answer as best I can!
>I first put instances on 1,000 random frames (total) from 55 videos. 

Do you mean that you added 1,000 frames to the labeling suggestions? When we refer to an `Instance`, we usually mean an annotation of the animal. The Labeling Suggestions is just a way to keep your labeling workflow organized by keeping track of frames you have already labeled and frames you plan to label.

>I then trained and fixed 250 predictions and would train again, because fixing 1,000 predictions seemed excessive.

Agreed, fixing 1,000 predictions per training round is indeed excessive. Usually the number of annotations you add (by fixing predictions) scales with the number of frames you have already annotated. As an example, if I am just starting a project, I would only label 5-20 frames and then do an initial round of training. This is to get some "rough" predictions out which can then be corrected faster than labeling from scratch. Then, for subsequent rounds of training I might only label in 20 frame increments until I get to 100 frames where I would increase to 50 frame increments. The idea here is that the more pre-existing annotations there are in your project, the more examples you would need to learn something new.

>However, I noticed that old predictions seemed to stay in addition to the new predictions with each trained model. 

I would expect the old predictions to stay if you are not using the [Labeling Suggestions in your workflow](https://sleap.ai/tutorials/initial-labeling.html). The workflow is as follows:
1. Initially, you would use the Labeling Suggestions dock to generate enough frames to label while leaving <however many frames you want to correct prediction on> unlabeled
2. Then you would run training, making sure to set Predict On: suggestions in the first tab of that dialog
3. You should now have prediction on your unlabeled frames in the Labeling Suggestions table which you can correct/convert the predictions to user-labeled instances to be used in the next round of training
4. Before doing another round of training, make sure to generate more labeled frames to the Labeling Suggestions, specifically, you would want <however many frames you want to correct prediction on>  generated
5. Repeat step 2-5

Sorry if I misunderstood. Let me know if that answered your question or not.

Thanks,
Liezl

**Comment by rhansen123** [comment]
Hello Liezl,

I understand completely now! Your explanation that was specific to my
question was perfect, and I feel I have a better grasp at training the
model needed for our experiment. Thank you for your help and timely
response!

Thank you again,
Rachael

On Fri, Oct 20, 2023 at 8:39 AM Liezl Maree ***@***.***>
wrote:

> Sorry, I am a bit confused - I think there might be a jargon mix up here,
> but I'll try to answer as best I can!
>
> I first put instances on 1,000 random frames (total) from 55 videos.
>
> Do you mean that you added 1,000 frames to the labeling suggestions? When
> we refer to an Instance, we usually mean an annotation of the animal. The
> Labeling Suggestions is just a way to keep your labeling workflow organized
> by keeping track of frames you have already labeled and frames you plan to
> label.
>
> I then trained and fixed 250 predictions and would train again, because
> fixing 1,000 predictions seemed excessive.
>
> Agreed, fixing 1,000 predictions per training round is indeed excessive.
> Usually the number of annotations you add (by fixing predictions) scales
> with the number of frames you have already annotated. As an example, if I
> am just starting a project, I would only label 5-20 frames and then do an
> initial round of training. This is to get some "rough" predictions out
> which can then be corrected faster than labeling from scratch. Then, for
> subsequent rounds of training I might only label in 20 frame increments
> until I get to 100 frames where I would increase to 50 frame increments.
> The idea here is that the more pre-existing annotations there are in your
> project, the more examples you would need to learn something new.
>
> However, I noticed that old predictions seemed to stay in addition to the
> new predictions with each trained model.
>
> I would expect the old predictions to stay if you are not using the Labeling
> Suggestions in your workflow
> <https://urldefense.com/v3/__https://sleap.ai/tutorials/initial-labeling.html__;!!K-Hz7m0Vt54!hquTm8rN9CbEQ_oxjYpmuUdSMTYnMKnfE8Bgp2asoFteVoVGohuoQ_mkNnyAfUFEPX8c8Oy7m7bR6H53CVzs9qI$>.
> The workflow is as follows:
>
>    1. Initially, you would use the Labeling Suggestions dock to generate
>    enough frames to label while leaving unlabeled
>    2. Then you would run training, making sure to set Predict On:
>    suggestions in the first tab of that dialog
>    3. You should now have prediction on your unlabeled frames in the
>    Labeling Suggestions table which you can correct/convert the predictions to
>    user-labeled instances to be used in the next round of training
>    4. Before doing another round of training, make sure to generate more
>    labeled frames to the Labeling Suggestions, specifically, you would want
>    generated
>    5. Repeat step 2-5
>
> Sorry if I misunderstood. Let me know if that answered your question or
> not.
>
> Thanks,
> Liezl
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/talmolab/sleap/discussions/1564*discussioncomment-7340630__;Iw!!K-Hz7m0Vt54!hquTm8rN9CbEQ_oxjYpmuUdSMTYnMKnfE8Bgp2asoFteVoVGohuoQ_mkNnyAfUFEPX8c8Oy7m7bR6H53I0VWp00$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/BDE6X26RGCWO5QUQGB65A5TYAKLKNAVCNFSM6AAAAAA6IAX7KOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TGNBQGYZTA__;!!K-Hz7m0Vt54!hquTm8rN9CbEQ_oxjYpmuUdSMTYnMKnfE8Bgp2asoFteVoVGohuoQ_mkNnyAfUFEPX8c8Oy7m7bR6H53zvkkHm0$>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Rachael Hansen
University of Washington - Heshmati/Golden Lab
Lab Technician
***@***.***
